assets/scss/components/_button.scss

```diff
@@ -59,6 +59,34 @@
     padding: 0.75rem 1rem;
     margin: 0;
     cursor: pointer;
+
+    &--with-icon {
```

Comment:
just `&--icon` would work as well, no need to overcomplicating the naming :)

Reply:
there is another c-button--icon classname specified. i think we can keep c-button--with-icon